### PR TITLE
feat(cli): improve the resource add command to generate page files for Next.js

### LINKED ISSUE #6091: [FEAT] To simplify adding a page after adding resources using CLI with Next.js.
### Is your feature request related to a problem? Please describe.

I usually use Refine with Next.js (App Router). The CLI command for adding resources is very convenient.

However, the files generated by the CLI are all components. Each time, I go through the following cumbersome process to convert them into pages:

1. Execute the npx refine add resource command.
    - This generates CRUD operation components under `/src/components/{resource name}`.
    - image
2. Move each .tsx file from `/src/components/{resource name}` to `/src/app/{resource name}`.
3. Modify the directory structure in `/src/app/{resource name}` according to the specifications in _refine_context.tsx. Rename each .tsx file moved in the previous step to page.tsx.
    - image
4. Replace each `export const` in the CRUD pages with `export default function`;
    - before: `export const UsersShow = () => {`
    - after: `export default function UsersShow() {`
    - If I use `export const`, the following error will be displayed in the browser when I open the page.
      - `Error: The default export is not a React Component in page: "/users/show/[id]"`

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Describe the thing to improve

Below is my request
1. When using Refine with Next.js, adding resources via CLI should generate appropriate page.tsx files under /src/app/.
2. Each generated page.tsx file should export the page component with `export default function`.

### PR DESCRIPTION
**Pull request by noritsune**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://refine.dev/docs/guides-concepts/contributing/#commit-convention

## Bugs / Features

- [x] Related issue(s) linked
- [x] Tests for the changes have been added
- [ ] Docs have been added / updated
- [x] Changesets have been added https://refine.dev/docs/guides-concepts/contributing/#creating-a-changeset

## What is the current behavior?
The files generated by the CLI are all components. So we have to make pages by hand to use it.

## What is the new behavior?
When using the add resource command in a project using Next.js, a page will be generated to perform the selected actions for that resource.

These pages simply display generated components that perform actions on the resource. The placement of page files assumes operation with the App Router. If you prefer to use the Page Router instead, you'll need to move them manually.

fixes #6091

### CODE REVIEW

**Review by aliemir — COMMENTED** [review]
Hey @noritsune, happy to see your contribution! Thank you for the PR! 🚀 We'll review it soon 🙏

**Comment by noritsune** [comment]
the test is failed... I'll check.

**Comment by noritsune** [comment]
Of couase, it passes on my pc.

**Comment by noritsune** [comment]
I changed the test code. It should passes.

File: packages/cli/templates/resource/components/create.tsx.hbs
Comment (alicanerdurmaz) [inline packages/cli/templates/resource/components/create.tsx.hbs:1]: if the project uses Next.js, the "use client" directive must be added.

File: packages/cli/templates/resource/components/edit.tsx.hbs
Comment (alicanerdurmaz) [inline packages/cli/templates/resource/components/edit.tsx.hbs:1]: if the project uses Next.js, the "use client" directive must be added.

File: packages/cli/templates/resource/components/list.tsx.hbs
Comment (alicanerdurmaz) [inline packages/cli/templates/resource/components/list.tsx.hbs:1]: if the project uses Next.js, the "use client" directive must be added.

File: packages/cli/templates/resource/components/show.tsx.hbs
Comment (alicanerdurmaz) [inline packages/cli/templates/resource/components/show.tsx.hbs:1]: if the project uses Next.js, the "use client" directive must be added.

**Review by alicanerdurmaz — CHANGES_REQUESTED** [review]

**Comment by alicanerdurmaz** [comment]
Hello @noritsune,

The PR looks great. I left a comment. If you could make that improvement too, it would be great. Thanks! 🙌

**Comment by noritsune** [comment]
Hello @alicanerdurmaz,
Thanks for the comments. I forgot to add "use client".
Now, I added them. Could you please check those page templates again?

File: packages/cli/templates/resource/pages/next/create/page.tsx.hbs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"use client";
```
Comment (alicanerdurmaz) [inline packages/cli/templates/resource/pages/next/create/page.tsx.hbs:1]: 'use client' should not be in page files, we should add it to component files(e.g., packages/cli/templates/resource/components/show.tsx.hbs)

Additionally, we should only add 'use client' only for Next.js projects

**Review by alicanerdurmaz — APPROVED** [review]

File: packages/cli/templates/resource/pages/next/create/page.tsx.hbs (hunk 1 above)
Comment (alicanerdurmaz) [inline packages/cli/templates/resource/pages/next/create/page.tsx.hbs:1]: 'use client' should not be in page files, we should add it to component files(e.g., packages/cli/templates/resource/components/show.tsx.hbs)

Additionally, we should only add 'use client' only for Next.js projects

**Review by alicanerdurmaz — CHANGES_REQUESTED** [review]

File: packages/cli/templates/resource/pages/next/create/page.tsx.hbs (hunk 1 above)
Comment (noritsune) [reply packages/cli/templates/resource/pages/next/create/page.tsx.hbs:1]: I misunderstood.
Indeed, "use client" should be added to the component file, not the page file. And it should only be added if you are using Next.js.
I will make the correction.

**Comment by noritsune** [comment]
@alicanerdurmaz I have modified the component template and added tests. Could you please review it again?

File: packages/cli/templates/resource/pages/next/create/page.tsx.hbs (hunk 1 above)
Comment (alicanerdurmaz) [reply packages/cli/templates/resource/pages/next/create/page.tsx.hbs]: @noritsune Thanks for everything! It looks great. I'll review by Monday at the latest. 🙌

**Review by aliemir — APPROVED** [review]

**Comment by noritsune** [comment]
The job `Check TSDoc Links` is failed...
I don't think this has anything to do with my commits. Or is it?

**Review by alicanerdurmaz — APPROVED** [review]

**Comment by alicanerdurmaz** [comment]
> The job `Check TSDoc Links` is failed... I don't think this has anything to do with my commits. Or is it?

it was probably not related to your commits, I updated the branch and now there is no problem 👍

**Comment by noritsune** [comment]
@alicanerdurmaz Thank you for the approve! My work is done.

**Pull request merged**